app/services/conversations/reply_service.rb

```diff
@@ -0,0 +1,59 @@
+class Conversations::ReplyService
+  def initialize(conversation)
+    @conversation = conversation
+  end
+
+  def can_reply?
+    channel = @conversation.inbox&.channel
+
+    return can_reply_on_instagram_via_messenger? if instagram_via_messenger?
```

Comment:
lets rewrite this to something like this 

```
return true unless channel&.messaging_window_enabled?

last_message_in_messaging_window?(messaging_window)
```

```
def messaging_window
   case @inbox.channel_type
    when 'Channel::Api'
       method_returning_api_channel_message_window
    when 'Channel::Whatsapp'
       24
    when 'Channel::Instagram'
      method returning instagram time
     when 'channel:: faceb
    end
end
```

lets have conditions for each inbox type explicitly and have spec for all the cases